Only add prediction to input line when there is some input

aperezdc opened the pull request:
Personally I like the line input prediction feature very useful, but I find it confusing that there is a suggestion on empty input lines. This patch disables showing any prediction on empty input.

I have checked how other shells do this, and for example Zsh with [zsh-autosuggestions](https://github.com/zsh-users/zsh-autosuggestions) ([screencast](https://asciinema.org/a/37390)) and Fish ([screencast](https://asciinema.org/a/183906)) do not show suggestions when the input is empty.

----

P.S: @magicant, it looks like the main repository for Yash is the Subversion one, so please let me know if you prefer contributions to be sent in some other way instead of GitHub PRs.

aperezdc committed:
Only add prediction to input line when there is some input

magicant commented:
Thank you for the patch, @aperezdc .

I know the current behavior can be confusing, but I still find it useful when entering a couple of commands routinely. I'm thinking of adding an option so the user can choose whichever behavior they like.

aperezdc commented:
@magicant Maybe I can try to add the option myself tomorrow. Is `lepredictempty` a good name for it?

magicant commented:
Yes, sounds pretty good 👍

aperezdc committed:
Add option "lepredictempty" to enable auto-suggestions on empty lines

Enabling the "lepredictempty" option will also provide
auto-suggestions on empty input lines when "lepredict" is
enabled.

aperezdc committed:
completion/set: Add "lepredictempty" to the list of options

aperezdc committed:
Document option "lepredictempty"

aperezdc committed:
tests: Modify test expectations to cover "lepredictempty"

This adds the "lepredictempty" option in the results for
some of the tests.

aperezdc commented:
@magicant I have added the option, and updated the documentation and the tests as well. I hope the PR looks better now :smiley:

magicant commented:
Thank you so much. You've done almost everything!

I merged your patch at a429dc9f35e1dd5594ec1675087a9cdc85136dab, and will follow up with an update to the Japanese documentation.

aperezdc commented:
@magicant Thanks for applying the patch and for taking care of the Japanese translation. Unfortunately I do not know enough Japanese myself :innocent:
